.travis.yml

```diff
@@ -1,29 +1,42 @@
+sudo: required
+dist: trusty
 language: c
 compiler:
     - clang
     - gcc
-before_install:
-    - sudo apt-get update -qq
-    - sudo apt-get install -qq libncursesw5-dev
-    - sudo apt-get install -qq libpulse-dev
-    - sudo apt-get install -qq libasound2-dev
-    - sudo apt-get install -qq libroar-dev
-    - sudo apt-get install -qq libao-dev
-    - sudo apt-get install -qq libjack-dev
-    - sudo apt-get install -qq libsamplerate0-dev
-    - sudo apt-get install -qq libcdio-dev
-    - sudo apt-get install -qq libcdio-cdda-dev
-    - sudo apt-get install -qq libcddb2-dev
-    - sudo apt-get install -qq libcue-dev
-#   - sudo apt-get install -qq libopus-dev
-    - sudo apt-get install -qq libflac-dev
-    - sudo apt-get install -qq libvorbis-dev
-    - sudo apt-get install -qq libmpcdec-dev
-    - sudo apt-get install -qq libwavpack-dev
-    - sudo apt-get install -qq libmad0-dev
-    - sudo apt-get install -qq libavformat-dev libavcodec-dev
-    - sudo apt-get install -qq libfaad-dev
-    - sudo apt-get install -qq libmp4v2-dev
-    - sudo apt-get install -qq libmodplug-dev
-    - sudo apt-get install -qq libmikmod2-dev
+install:
+    - if [ "$CC" = "gcc" ]; then export CC="gcc-6"; fi
+    - if [ "$CC" = "clang" ]; then export CC="clang-3.8"; fi
+addons:
+    apt:
+        sources:
+            - ubuntu-toolchain-r-test
+            - sourceline: 'deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-3.8 main'
+            - sourceline: 'deb-src http://apt.llvm.org/trusty/ llvm-toolchain-trusty-3.8 main'
+        packages:
+            - gcc-6
+            - clang-3.8
```

Comment:
These are pretty bleeding edge versions. We should probably use something older in Travis, so that we will not get situations like «it compiles in Travis, but fails in SOME_POPULAR_DISTRO».